zh/deploy-failures.md

```diff
@@ -0,0 +1,86 @@
+---
+title: 部署错误
+summary: 介绍 Kubernetes 上 TiDB 部署的常见错误以及处理办法。
+category: how-to
+---
+
+# Kubernetes 上的 TiDB 常见部署错误
+
+本文介绍了 Kubernetes 上 TiDB 常见部署错误以及处理办法。
+
+## Pod 未正常创建
+
+创建集群后，如果 Pod 没有创建，则可以通过以下方式进行诊断：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl get tidbclusters -n ${namespace}
+kubectl get statefulsets -n ${namespace}
+kubectl describe statefulsets -n ${namespace} ${cluster_name}-pd
+```
+
+## Pod 处于 Pending 状态
+
+Pod 处于 Pending 状态，通常都是资源不满足导致的，比如：
+
+* 使用持久化存储的 PD、TiKV、Monitor Pod 使用的 PVC 的 StorageClass 不存在或 PV 不足
+* Kubernetes 集群中没有节点能满足 Pod 申请的 CPU 或内存
+* PD 或者 TiKV Replicas 数量和集群内节点数量不满足 tidb-scheduler 高可用调度策略
+
+此时，可以通过 `kubectl describe pod` 命令查看 Pending 的具体原因：
+
+{{< copyable "shell-regular" >}}
+
+```
+kubectl describe po -n ${namespace} ${pod_name}
+```
+
+如果是 CPU 或内存资源不足，可以通过降低对应组件的 CPU 或内存资源申请使其能够得到调度，或是增加新的 Kubernetes 节点。
+
+如果是 PVC 的 StorageClass 找不到，需要将 `storageClassName` 修改为集群中可用的 StorageClass 名字，然后运行 `kubectl apply -f tidb-cluster.yaml` 进行集群更新，然后将 Statefulset 删除，并且将对应的 PVC 也都删除，可以通过以下命令获取集群中可用的 StorageClass：
+
+{{< copyable "shell-regular" >}}
+
+```
+kubectl get storageclass
+```
+
+如果集群中有 StorageClass，但可用的 PV 不足，则需要添加对应的 PV 资源。对于 Local PV，可以参考[本地 PV 配置](configure-storage-class.md#本地-pv-配置)进行扩充。
+
+tidb-scheduler 针对 PD 和 TiKV 定制了高可用调度策略。对于同一个 TiDB 集群，假设 PD 或者 TiKV 的 Replicas 数量为 N，那么可以调度到每个节点的 PD Pod 数量最多为 `M=(N-1)/2`（如果 N<3，M=1），可以调度到每个节点的 TiKV Pod 数量最多为 `M=ceil(N/3)`（ceil 表示向上取整，如果 N<3，M=1）。如果 Pod 因为不满足高可用调度策略而导致状态为 Pending，需要往集群内添加节点。
```

Comment:
```suggestion
### 可用 PV 不足

如果集群中有 StorageClass，但可用的 PV 不足，则需要添加对应的 PV 资源。对于 Local PV，可以参考[本地 PV 配置](configure-storage-class.md#本地-pv-配置)进行扩充。

### 不满足 tidb-scheduler 高可用策略

tidb-scheduler 针对 PD 和 TiKV 定制了高可用调度策略。对于同一个 TiDB 集群，假设 PD 或者 TiKV 的 Replicas 数量为 N，那么可以调度到每个节点的 PD Pod 数量最多为 `M=(N-1)/2`（如果 N<3，M=1），可以调度到每个节点的 TiKV Pod 数量最多为 `M=ceil(N/3)`（ceil 表示向上取整，如果 N<3，M=1）。

如果 Pod 因为不满足高可用调度策略而导致状态为 Pending，需要往集群内添加节点。
```

Reply:
@weekface I add level-3 headings to each cause/solution above, so users can easily locate their problems according to the **navigation bar** on the right side of our website.

Please apply these changes to other parts of this doc. Thanks~

Reply:
done